--- openmpi-1.8.3/config/opal_functions.m4.orig	2014-07-11 20:12:32.000000000 +0200
+++ openmpi-1.8.3/config/opal_functions.m4	2014-11-24 20:46:32.199412068 +0100
@@ -473,6 +473,11 @@
 
 ompi_prog=$1
 
+case "$omni_prog" in
+/*)
+	$2=$omni_prog
+	;;
+*)
 IFS_SAVE=$IFS
 IFS="$PATH_SEPARATOR"
 for ompi_dir in $PATH; do
@@ -482,6 +487,8 @@
     fi
 done
 IFS=$IFS_SAVE
+;;
+esac
 
 OPAL_VAR_SCOPE_POP
 ])dnl
