--- openmpi-1.2.5/./config/ompi_get_version.sh~	2008-01-09 01:37:36.000000000 +0200
+++ openmpi-1.2.5/./config/ompi_get_version.sh	2008-01-09 01:37:42.931463912 +0200
@@ -59,7 +59,7 @@
     : ${svnversion_result=-1}
 
         if test -f "$srcfile"; then
-        eval "`sed -n \"\
+        eval $(sed -n '
 	t clear
 	: clear
 	s/^major/OMPI_MAJOR_VERSION/
@@ -71,7 +71,7 @@
 	t print
 	b
 	: print
-	p\" < \"\$srcfile\"`"
+	p' < "$srcfile")
 
         # Only print release version if it isn't 0
         if test $OMPI_RELEASE_VERSION -ne 0 ; then
--- openmpi-1.2.5/./config/ompi_get_version.m4~	2006-08-15 23:14:05.000000000 +0300
+++ openmpi-1.2.5/./config/ompi_get_version.m4	2008-01-09 01:38:42.551027180 +0200
@@ -41,7 +41,7 @@
 
     dnl quote eval to suppress macro expansion with non-GNU m4
     if test -f "$1"; then
-        [eval] "`sed -n \"\
+        [eval] $(sed -n '
 	t clear
 	: clear
 	s/^major/$2_MAJOR_VERSION/
@@ -53,7 +53,7 @@
 	t print
 	b
 	: print
-	p\" < \"\$1\"`"
+	p' < "$1")
 
         # Only print release version if it isn't 0
         if test $$2_RELEASE_VERSION -ne 0 ; then
